smartcontract/storage/cachedb.go

```diff
@@ -117,24 +117,47 @@ func (self *CacheDB) GetContract(addr comm.Address) (*payload.DeployCode, error)
 	return contract, nil
 }
 
-func (self *CacheDB) PutContract(contract *payload.DeployCode) error {
+func (self *CacheDB) PutContract(contract *payload.DeployCode) {
 	address := contract.Address()
 
 	sink := comm.NewZeroCopySink(0)
-	err := contract.Serialization(sink)
-	if err != nil {
-		return err
-	}
+	contract.Serialization(sink)
 
 	value := sink.Bytes()
 	self.put(common.ST_CONTRACT, address[:], value)
-	return nil
 }
 
 func (self *CacheDB) DeleteContract(address comm.Address) {
 	self.delete(common.ST_CONTRACT, address[:])
 }
 
+func (self *CacheDB) GetMetaData(addr comm.Address) (*payload.MetaDataCode, error) {
+	value, err := self.get(common.ST_CONTRACT_META_DATA, addr[:])
+	if err != nil {
+		return nil, err
+	}
+
+	if len(value) == 0 {
+		return nil, nil
```

Comment:
return default-metadata?

Reply:
use meta data == nil to judge is meta data initialized